java/modules/core/src/main/java/org/apache/synapse/registry/url/SimpleURLRegistry.java

```diff
@@ -60,9 +61,10 @@ public OMNode lookup(String key) {
             log.debug("==> Repository fetch of resource with key : " + key);
 
         }
-        URL url = SynapseConfigUtils.getURLFromPath(root + key, properties.get(
+        @SuppressWarnings("startswith") URL url = SynapseConfigUtils.getURLFromPath(root + key, properties.get(
                 SynapseConstants.SYNAPSE_HOME) != null ?
                 properties.get(SynapseConstants.SYNAPSE_HOME).toString() : "");
+        //TRUE POSITIVE: It is possible that http creeps in cause we don't know how root is changed
```

Comment:
Is there somewhere in particular that root can be changed to something other than an acceptable String?